android/src/main/java/com/baseflow/permissionhandler/PermissionUtils.java

```diff
@@ -0,0 +1,31 @@
+package com.baseflow.permissionhandler;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+
+import androidx.annotation.RequiresApi;
+import androidx.core.app.ActivityCompat;
+
+public class PermissionUtils {
+
+  @RequiresApi(api = Build.VERSION_CODES.M)
+   static boolean neverAskAgainSelected(final Activity activity, final String permission) {
+    final boolean prevShouldShowStatus = getRationaleDisplayStatus(activity, permission);
```

Comment:
Variable name suggests it holds a result of a previous call to the `shouldShowRequestPermissionRationale`  while in fact it holds information if the App has requested the particular permission before. I suggest you rename the variable to something like:
```suggestion
    final boolean hasRequestedPermissionBefore = getRequestedPermissionBefore(activity, permission);
```